Source/RSP/RSP Command.c

```diff
@@ -1238,59 +1279,28 @@ char * RSPSc2Name ( DWORD OpCode, DWORD PC )
 
 	PC = PC; // unused
 
-	switch (command.rd)
+	if (strcmp(mnemonics_swc2[command.rd], unused_op) == 0)
 	{
-	case RSP_LSC2_BV:
-		sprintf(CommandName,"SBV\t$v%d [%d], 0x%04X (%s)",command.rt, command.del, 
-			command.voffset, GPR_Name(command.base));
-		break;
-	case RSP_LSC2_SV:
-		sprintf(CommandName,"SSV\t$v%d [%d], 0x%04X (%s)",command.rt, command.del, 
-			(command.voffset << 1), GPR_Name(command.base));
-		break;
-	case RSP_LSC2_LV:
-		sprintf(CommandName,"SLV\t$v%d [%d], 0x%04X (%s)",command.rt, command.del, 
-			(command.voffset << 2), GPR_Name(command.base));
-		break;
-	case RSP_LSC2_DV:
-		sprintf(CommandName,"SDV\t$v%d [%d], 0x%04X (%s)",command.rt, command.del, 
-			(command.voffset << 3), GPR_Name(command.base));
-		break;
-	case RSP_LSC2_QV:
-		sprintf(CommandName,"SQV\t$v%d [%d], 0x%04X (%s)",command.rt, command.del, 
-			(command.voffset << 4), GPR_Name(command.base));
-		break;
-	case RSP_LSC2_RV:
-		sprintf(CommandName,"SRV\t$v%d [%d], 0x%04X (%s)",command.rt, command.del, 
-			(command.voffset << 4), GPR_Name(command.base));
-		break;
-	case RSP_LSC2_PV:
-		sprintf(CommandName,"SPV\t$v%d [%d], 0x%04X (%s)",command.rt, command.del, 
-			(command.voffset << 3), GPR_Name(command.base));
-		break;
-	case RSP_LSC2_UV:
-		sprintf(CommandName,"SUV\t$v%d [%d], 0x%04X (%s)",command.rt, command.del, 
-			(command.voffset << 3), GPR_Name(command.base));
-		break;
-	case RSP_LSC2_HV:
-		sprintf(CommandName,"SHV\t$v%d [%d], 0x%04X (%s)",command.rt, command.del, 
-			(command.voffset << 4), GPR_Name(command.base));
-		break;
-	case RSP_LSC2_FV:
-		sprintf(CommandName,"SFV\t$v%d [%d], 0x%04X (%s)",command.rt, command.del, 
-			(command.voffset << 4), GPR_Name(command.base));
-		break;
-	case RSP_LSC2_WV:
-		sprintf(CommandName,"SWV\t$v%d [%d], 0x%04X (%s)",command.rt, command.del, 
-			(command.voffset << 4), GPR_Name(command.base));
-		break;
-	case RSP_LSC2_TV:
-		sprintf(CommandName,"STV\t$v%d [%d], 0x%04X (%s)",command.rt, command.del, 
-			(command.voffset << 4), GPR_Name(command.base));
-		break;
-	default:
-		sprintf(CommandName,"RSP: Unknown\t%02X %02X %02X %02X",
-			command.Ascii[3],command.Ascii[2],command.Ascii[1],command.Ascii[0]);
+		sprintf(
+			CommandName,
+			"RSP: Unknown\t%02X %02X %02X %02X",
+			command.Ascii[3],
+			command.Ascii[2],
+			command.Ascii[1],
+			command.Ascii[0]
+		);
+	}
+	else
+	{
+		sprintf(
+			CommandName,
+			"%s\t$v%d[%d], 0x%04X(%s)",

```

Comment:
Here is where we can start to document actual opcode formats to the RSP.
```c
"%s\t$%d, $%d, $%d" // R-type instructions:  rd, rs, rt
"%s\t$%d, $%d, 0x%04X" // I-type instructions:  rt, rs, immediate
"%s\t0x%07X" // J-type instructions:  (target << 2)
"%s\t$v%d, $v%d, $v%d[%s]" // most vector unit computational instructions
"%s\t$%d, 0x%03X" // I-type, register-immediate RSP-encoded
...
```
These are just some of the ideas @project64 that I think can be worth warranting a discussion in the future.